dali/kernels/audio/mel_scale/mel_filter_bank_gpu.cu

```diff
@@ -205,19 +462,43 @@ class MelFilterBankGpu<T>::Impl : public MelFilterImplBase<T> {
     nframes_.clear();
     nwindows_.clear();
     block_descs_.clear();
+    block2sample_.clear();
     auto batch_size = in_shape.num_samples();
+
+    int max_shm_size = GetSharedMemPerBlock();
+    for (shm_height_ = 32; ; shm_height_ >>= 1) {
+      int shm_in = mel_inner_fft::shm_in_size<T>(shm_height_);
+      int shm_out = mel_inner_fft::shm_out_size<T>(shm_height_, args_.nfilter);
+      shm_size_ = shm_in + shm_out;
+      if (shm_size_ <= max_shm_size)
+        break;
+    }
+
+    if (shm_height_ < 1)
+      throw std::out_of_range(make_string("Too many Mel filters ", args_.nfilter));
+
```

Comment:
What's the limit? Should we document it somehow in the op documentation?

Reply:
The limit is high (thousands of Mel filters) and depends on axis (we have no limit on outer FFT) and on architecture (shared memory size). Putting a formula there seems a bit excessive. I can calculate the limit here, so we can print what the max. is if it's exceeded.